### ISSUE TITLE
Windows cmd terminal doesn't per default enable ANSI code interpretation

### ISSUE THREAD

**Issue by sewbacca** [hoelzro/lua-term#23]
If using ANSI codes inside the cmd terminal, they will simply be printed as text. However they can be enabled, either quite hackily with `os.execute ""` or via the official Win API. I stumbled accross this issue, when the codes were not working at all inside my VSCode integreated terminal. So I went down a rabbit whole and found [this stackoverflow question](https://stackoverflow.com/questions/64919350/enable-ansi-sequences-in-windows-terminal) which perfectly explains why `os.execute ""` fixes the issue and how to use the official Win API to achieve the same result.

**Comment by Tieske** [comment]
Here's a luasystem PR that adds this: https://github.com/lunarmodules/luasystem/pull/21

**Comment by Tieske** [comment]
the above mentioned PR has been merged an luasystem 0.4.0 has been released with that functionality.

@hoelzro would you accept a PR that tests for the presence of that functionality, and automatically enables it if found?

Something like this:
```lua
do
  local ok, sys = pcall(require, "system")
  if ok and sys.setconsoleflags and sys.isatty(io.stdout) then
    -- setup Windows console to handle ANSI processing
    sys.setconsoleflags(io.stdout, sys.getconsoleflags(io.stdout) + sys.COF_VIRTUAL_TERMINAL_PROCESSING)
    -- set output to UTF-8
    sys.setconsoleoutputcp(sys.CODEPAGE_UTF8)
  end
end

**Comment by hoelzro** [comment]
@Tieske Sure!
